disnake/guild.py

```diff
@@ -1974,16 +1974,36 @@ async def edit(
             fields["system_channel_flags"] = system_channel_flags.value
 
         if community is not MISSING:
-            features = []
+            # if we don't have complete feature information for the guild
+            # it is possible to disable or enable other features that we didn't intend to touch
+            # The issue is removing community also removes partner status, and discord is adding the ability to
+            # enable or disable invite usage for guilds, which is problematically triggered by a feature.
+            # To enable or disable that feature, we will need to provide all of the existing features in advance.
+            # That is problematic to say the least. By reimplementing this list to only provide the features if we make a change
+            # we minimize the possible times that we will provide features when no changes are made, and possibly affect other features.
+            if self.unavailable:
+                raise RuntimeError(
+                    "cannot modify features of an unavailable guild due to potentially destructive results."
+                )
+            features = self.features
```

Comment:
this mutates the instance's `features` list, should either copy or make it a set which would also simplify some code below

Reply:
https://github.com/DisnakeDev/disnake/pull/705/commits/badbf3f29b225c6bb274f88a910581c4f6c15eb8